drv/gimlet-seq-server/src/main.rs

```diff
@@ -103,451 +104,381 @@ enum Trace {
     SpdBankAbsent(u8),
     SpdAbsent(u8, u8, u8),
     SpdDimmsFound(usize),
-
+    I2cFault {
+        retries_remaining: u8,
+        code: i2c::ResponseCode,
+    },
+    StartFailed(SeqError),
     None,
 }
 
 ringbuf!(Trace, 128, Trace::None);
 
 #[export_name = "main"]
 fn main() -> ! {
-    let spi = drv_spi_api::Spi::from(SPI.get_task_id());
     let sys = sys_api::Sys::from(SYS.get_task_id());
     let jefe = Jefe::from(JEFE.get_task_id());
+    let spi = drv_spi_api::Spi::from(SPI.get_task_id());
     let hf = hf_api::HostFlash::from(HF.get_task_id());
+    match ServerImpl::init(&sys, &jefe, spi, hf) {
+        // Set up everything nicely, time to start serving incoming messages.
+        Ok(mut server) => {
+            let mut buffer = [0; idl::INCOMING_SIZE];
+            loop {
+                idol_runtime::dispatch(&mut buffer, &mut server);
+            }
+        }
 
-    // Ensure the SP fault pin is configured as an open-drain output, and pull
-    // it low to make the sequencer restart externally visible.
-    sys.gpio_configure_output(
-        FAULT_PIN_L,
-        sys_api::OutputType::OpenDrain,
-        sys_api::Speed::Low,
-        sys_api::Pull::None,
-    );
-    sys.gpio_reset(FAULT_PIN_L);
-
-    // Turn off the chassis LED, in case this is a task restart (and not a
-    // full chip restart, which would leave the GPIO unconfigured).
-    sys.gpio_configure_output(
-        CHASSIS_LED,
-        sys_api::OutputType::PushPull,
-        sys_api::Speed::Low,
-        sys_api::Pull::None,
-    );
-    sys.gpio_reset(CHASSIS_LED);
-
-    // To allow for the possibility that we are restarting, rather than
-    // starting, we take care during early sequencing to _not turn anything
-    // off,_ only on. This means if it was _already_ on, the outputs should not
-    // glitch.
-
-    // Unconditionally set our power-good detects as inputs.
-    //
-    // This is the expected reset state, but, good to be sure.
-    sys.gpio_configure_input(PGS_PINS, PGS_PULL);
-
-    // Set SP3_TO_SP_NIC_PWREN_L to be an input
-    sys.gpio_configure_input(NIC_PWREN_L_PINS, NIC_PWREN_L_PULL);
+        // Initializing the sequencer failed.
+        Err(_) => {
+            // Tell everyone that something's broken, as loudly as possible.
+            ringbuf_entry!(Trace::StartFailed(SeqError::I2cFault));
+            sys.gpio_set(FAULT_PIN_L);
+            jefe.set_state(PowerState::InitFailed as u32);
+
+            // All these moments will be lost in time, like tears in rain...
+            // Time to die.
+            loop {
+                hl::sleep_for(100);
+            }
```

Comment:
I wasn't sure what the most Good and Proper way for a Hubris task to "die permanently" is --- is there something better to do here?

Reply:
Waiting for a notification that will never come would be nicer as this should make sure the task never becomes ready to run again (e.g. https://github.com/oxidecomputer/hubris/blob/master/task/sp_measure/src/main.rs#L71-L75)

Reply:
Lovely, thank you!